### ISSUE TITLE
RPM package should provide "docker" for compatibility

### ISSUE THREAD

**Issue by oszi** [moby/moby#26059]
**Output of `docker version`:**

**Output of `docker info`:**

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Steps to reproduce the issue:**

The Fedora and CentOS docker packages conflict with the upstream docker-engine package
but this package does not provide "docker" on which many packages depend.

**Describe the results you received:**

So for instance, `yum install kubernetes` will fail.

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

Basically, the RPM spec file needs `Provides: docker = %{version}-%{release}`
It would be great if this could be backported to older versions as well.

Bugzilla: [#1370628](https://bugzilla.redhat.com/show_bug.cgi?id=1370628)

**Comment by cpuguy83** [comment]
ping @tiborvass

**Comment by runcom** [comment]
Makes sense since it was docker at the beginning and that changed after

**Comment by thaJeztah** [comment]
@oszi perhaps you're interested in opening a pull-request for discussion? the RPM specs are in this file https://github.com/docker/docker/blob/master/hack/make/.build-rpm/docker-engine.spec

**Comment by oszi** [comment]
@thaJeztah How long do we have in the 1.13 milestone?
I can do it later this week or at the weekend.

**Comment by thaJeztah** [comment]
This week or weekend is ok, there's no fixed date set for 1.13 yet

**Comment by oszi** [comment]
PR opened. So this does not make the upstream docker-engine a perfect drop-in replacement for the downstream docker package because they have different systemd units and there are some downstream patches but it does work nicely with packages like atomic or skopeo that depend on docker.

The only thing I'm worried about is docker-engine-selinux and container-selinux not playing nice.
(ping @rhatdan)

**Comment by rhatdan** [comment]
docker-engine-selinux is downstream of container-selinux.  So as long as docker-engine rebases, their should be not problems.  If paths change, then it might need updates.

**Comment by thaJeztah** [comment]
Scripts for packaging are in https://github.com/docker/docker-ce-packaging (but doesn't have an issue tracker enabled)
